### ISSUE TITLE
functions: storage generic sample fails to deploy

### ISSUE THREAD

**Issue by eRadical** [GoogleCloudPlatform/golang-samples#750]
https://github.com/GoogleCloudPlatform/golang-samples/blob/b824bfcb4933d9f3ec98bcf790bb230ae4f01d26/functions/helloworld/storage_generic/hello_storage_generic.go#L16

I'm not able to deploy this function "as is" in this example.

Deploying function (may take a while - up to 2 minutes)...failed.                                                                                                                                                 
ERROR: (gcloud.functions.deploy) OperationError: code=3, message=Build failed: go: finding cloud.google.com/go/functions/metadata v0.0.0
import cycle not allowed
package serverlessapp
	imports cloud.google.com/go/functions/metadata
	imports cloud.google.com/go/functions/metadata

**Comment by tbpg** [comment]
Thank you for the bug report! I'm working on PR to add a `go.mod`, which should help with this.

Did you have a `go.mod` or `vendor` directory when you deployed? I haven't seen this error before - I want to understand what happened so I can make it more clear.

**Comment by eRadical** [comment]
Hi Tyler,

Last night @mihaioprea figured this out... Indeed a go.mod is required (I tried it but with wrong content that was pulling more than what was needed).

**Comment by tbpg** [comment]
See https://cloud.google.com/functions/docs/concepts/go-runtime#specifying_dependencies for docs about how to add dependencies.

TL;DR:
```
export GO111MODULE=on
go mod init
go mod tidy
```

If you're outside `GOPATH`, do `go mod init example.com/foo`.

**Comment by eRadical** [comment]
Awesome!
